sentry/src/main/java/io/sentry/Scopes.java

```diff
@@ -44,28 +41,27 @@ public final class Scopes implements IScopes, MetricsApi.IMetricsInterface {
   Scopes(
       final @NotNull IScope scope,
       final @NotNull IScope isolationScope,
-      final @NotNull SentryOptions options,
       final @NotNull String creator) {
-    this(scope, isolationScope, null, options, creator);
+    this(scope, isolationScope, null, creator);
   }
 
   private Scopes(
       final @NotNull IScope scope,
       final @NotNull IScope isolationScope,
       final @Nullable Scopes parentScopes,
-      final @NotNull SentryOptions options,
       final @NotNull String creator) {
-    validateOptions(options);
-
     this.combinedScope = new CombinedScopeView(getGlobalScope(), isolationScope, scope);
     this.scope = scope;
     this.isolationScope = isolationScope;
     this.parentScopes = parentScopes;
     this.creator = creator;
-    this.options = options;
+
+    final @NotNull SentryOptions options = getOptions();
+    validateOptions(options);
     this.tracesSampler = new TracesSampler(options);
     this.transactionPerformanceCollector = options.getTransactionPerformanceCollector();
 
+    // TODO [HSM] Checking isEnabled may not be what we want with global scope anymore
```

Comment:
Allowing users to use global scope before `Sentry.init` may imply that we can't noop if not yet enabled.

Reply:
I'd say if the SDK hasn't been enabled yet we shouldn't send anything. We shouldn't store things to send once enabled as the SDK might never be enabled.

RFC also suggests to not do anything in that case: https://github.com/getsentry/rfcs/pull/122/files#diff-c9abcee1a2612ddd7dd66e226e832f520e6b4cbc67bdcbb5371514af7d25ae38R94